cmd/update-main_test.go

```diff
@@ -158,6 +164,59 @@ func TestGetCurrentReleaseTime(t *testing.T) {
 	}
 }
 
+// Tests user agent string.
+func TestUserAgent(t *testing.T) {
+	customUserAgent := func(mode string, extraMode string) string {
```

Comment:
Because this function accepts switches, there is a possibility of breakage if wrong inputs are passed. Tabular cases and string comparison are clean and simple.